GANDLF/utils/generic.py

```diff
@@ -114,15 +99,21 @@ def version_check(version_from_config: str, version_to_check: str) -> bool:
     Returns:
         bool: If the version of the config file is compatible with the version of the code.
     """
-    version_to_check_int = parse_version(version_to_check)
-    min_ver = parse_version(version_from_config["minimum"])
-    max_ver = parse_version(version_from_config["maximum"])
-    if (min_ver > version_to_check_int) or (max_ver < version_to_check_int):
-        sys.exit(
-            "Incompatible version of GaNDLF detected ("
-            + str(version_to_check_int)
-            + ")"
-        )
+    version_to_check_int = Version(version_to_check)
+    min_ver = Version(version_from_config.get("minimum"), None)
+    max_ver = Version(version_from_config.get("maximum"), None)
```

Comment:
What is `None` as second arg here? don't see any second arg in the doc

Reply:
:+1:
Then you may simplify this: you don't need to pass `None` as default value explicitly:
```
In [5]: d = {"a": 2, "b": 123}

In [6]: d.get("c")

In [7]: str(d.get("c"))
Out[7]: 'None'

In [8]: d.get("c") is None
Out[8]: True
```

Reply:
This has changed. The `None` should be the default for `.get()`. Thanks for catching it!

Reply:
@scap3yvt this `Version(version_from_config.get("minimum", None))`, `.get` + `None` are excessive: if `minimum` version was not found in the config, you pass `None` and the code anyway fails because `Version()` cannot process it. So no need to guard from empty config as version is obligatory any way.
So, you may simplify it to just
```
min_ver = Version(version_from_config["minimum"])
max_ver = Version(version_from_config["maximum"])
```

Reply:
Got it, and I have made appropriate changes.

Reply:
Thanks a lot, man!